forch/faucet_state_collector.py

```diff
@@ -569,24 +571,74 @@ def _fill_path_to_root(self, switch_name, switch_map):
         """populate path to root for switch_state"""
         switch_map["root_path"] = self.get_switch_egress_path(switch_name)
 
-    def _fill_dva_states(self, switch_name, port_id, port_map):
-        dp_obj = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
-        if not dp_obj:
+    def _fill_vlan_behavior(self, switch_name, switch_map, metrics=None):
+        dp_config = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
+        if not dp_config:
             LOGGER.warning('Switch not defined in dps config: %s', switch_name)
             return
 
-        port_obj = dp_obj.interfaces.get(port_id)
-        if not port_obj:
+        vlans_map = switch_map.setdefault('vlans', {})
+        for vid, vlan_config in dp_config.vlans.items():
+            acl_maps_list = vlans_map.setdefault(int(vid), {}).setdefault('acls', [])
+            if metrics and 'flow_packet_count_vlan_acl' in metrics:
+                samples = metrics['flow_packet_count_vlan_acl'].samples
+            else:
+                samples = None
+            self._fill_acls_behavior(switch_name, acl_maps_list, vlan_config.acls_in, samples)
+
+    def _fill_port_behavior(self, switch_name, port_id, port_map, metrics=None):
+        dp_config = self.faucet_config.get(DPS_CFG, {}).get(switch_name)
+        if not dp_config:
+            LOGGER.warning('Switch not defined in dps config: %s', switch_name)
+            return
+
+        port_config = dp_config.ports.get(port_id)
+        if not port_config:
             LOGGER.warning('Port not defined in dps config: %s, %s', switch_name, port_id)
             return
 
-        native_vlan = port_obj.get('native_vlan')
-        acls_in = port_obj.get('acls_in')
+        if port_config.native_vlan:
+            port_map['vlan'] = int(port_config.native_vlan.vid)
+        if port_config.acls_in:
+            acl_maps_list = port_map.setdefault('acls', [])
+            if metrics and 'flow_packet_count_port_acl' in metrics:
+                samples = metrics['flow_packet_count_port_acl'].samples
+            else:
+                samples = None
+            self._fill_acls_behavior(
+                switch_name, acl_maps_list, port_config.acls_in, samples, port_id)
+
+    def _fill_acls_behavior(self, switch_name, acls_map_list, acls_config,
+                            metric_samples=None, port_id=None):
+        for acl_config in acls_config:
+            acl_map = {'name': acl_config._id}
```

Comment:
where does ._id come from?  I can't find it referenced anywhere -- seems odd for a name (_ usually means it's private).

Reply:
It is of type "faucet/acl.py:class ACL(Conf)". 

"_id" is the only attribute field I can find to get the name.